main.tf

```diff
@@ -9,15 +17,11 @@ data "aws_ami" "sec" {
     values = ["x86_64"]
   }
 
-  owners      = ["692314432491"]
+  owners = ["692314432491"]
 
   most_recent = true
 }
 
-data "aws_vpc" "sec" {
```

Comment:
Nothing expect this?

Reply:
Nope, tflint spotted it. I was previously using this to restrict the security group on the EC2 instance to traffic just from the VPCs. But NLBs are weird; they don't have their own SGs, they rely on the SGs of the EC2 instances. So if I block external traffic from the EC2 instance, traffic from outside the VPC won't flow in. 

The EC2 instance is however on the private subnet, so even if it is open to the world, it is not possible to reach it unless you go through the NLB.